### ISSUE TITLE
General Project Status Update

### ISSUE THREAD

**Issue by pauldorehill** [rustwasm/wasm-bindgen#2337]
### Summary

Would the main contributors @alexcrichton & @fitzgen be able to say a few words around the general plans / status of the project?

### Additional Details

It has been suggested that the `rustwasm` working group is in a state of (semi?) stasis and:
- [last blog post](https://rustwasm.github.io) March '19
- it looks like there are no official team members [@rustwasm/core](https://github.com/orgs/rustwasm/teams/core/members)
- the [Mozilla layoffs](https://www.reddit.com/r/rust/comments/i8yfwj/much_of_the_rustwasmtime_team_hit_by_layoffs_at/)
- Alex the [main contributor](https://github.com/rustwasm/wasm-bindgen/graphs/contributors) announced his [reduced rust involvement](https://internals.rust-lang.org/t/scaling-back-my-involvement-in-rust/11754) but is still going strong merging pull requests and publishing updates 😄 .

`wasm-bindgen` is at the core of `rust` + `web` and is at the foundations of many great projects: [seed](https://github.com/seed-rs/seed/blob/master/Cargo.toml), [iced](https://github.com/hecrj/iced/blob/master/web/Cargo.toml), [dominator](https://github.com/Pauan/rust-dominator), [mogwai](https://github.com/schell/mogwai), [yew](https://github.com/yewstack/yew/blob/master/yew/Cargo.toml) to name but a few. 

I'm extremely thankful that this project exists and I'm not after reassurances or commitments: rather for all of us having the marvelous experience of running rust in the browser it would be nice to have a clear understanding of where things currently stand.

**Comment by alexcrichton** [comment]
I personally have been and plan to continue maintaining wasm-bindgen, primarily making sure small enough PRs get merged and bugs get fixed. I have less time/motivation these days for new features, however. I'm not personally developing new features and would find it difficult to review/collaborate on a large new feature.

I would be happy to invite others to help maintain this crate if they'd like, however!

The Rust and WebAssembly Working Group no longer has the vibrance it once had, so there's not necessarily a group of folks helping drive the vision for Rust and WebAssembly. I would be the first to list out the issues we currently have with the various rustwasm repositories we have, but unfortunately I'm not entirely sure of the best way forward. It would probably be good to get some folks on board with helping to maintain the book, but I suspect that wasm-bindgen won't be living inside of this GitHub organization indefinitely.

Perhaps the biggest thing I'd like to fix is better tooling around running the `wasm-bindgen` CLI. I don't think `wasm-pack` is living up to its promise and would love to have a replacement tool to suggest. Unfortunately though I don't have the time to pioneer this myself. I worked on [`cargo-wasi`](https://github.com/bytecodealliance/cargo-wasi) awhile back which I think might be a good template for how a wrapper tool might work, and I've reserved the [`cargo-wasm`](https://crates.io/crates/cargo-wasm) name on crates.io if needed, but in any case work needs to be done and we'd have to find someone to do it primarily!

**Comment by pauldorehill** [comment]
Alex,

Thanks for taking the time to provide a response.

> but I suspect that wasm-bindgen won't be living inside of this GitHub organization indefinitely.

I'd postulate that the time has come for this: the continued existence of the non existent _official rust_ `wasm-wg` is misleading to all those using and learning rust and wasm. Moving `wasm-bindgen` would be a clean way to move on from the current situation and provide a clear path for the future. 

I'm sure there will be some loud voices crying about such heresy, but I've a strong suspicion that there is a **silent and silenced majority** that would support your decision. Most of these people are more interested in pragmatic nature of code and are less likely to speak up than those interested in narrative control. There are very few that could stand toe to toe with you when it comes to your contributions to rust & wasm-bindgen. 

In regard to the `wasm-bindgen` / `wasm-pack` debate I actually think the current status is confusing: you can see that by conflation of issues in the repositories. The way `wasm-pack` is threaded through all the documentation it makes it seem that it is doing the heavy lifting: rather its roll is to sync the versions of `wasm-bindgen` + [wasm-bindgen-cli](https://rustwasm.github.io/docs/wasm-bindgen/reference/cli.html), then run the cli & [wasm-opt](https://github.com/WebAssembly/binaryen/releases). 

In an similar vain [the book](https://rustwasm.github.io/docs/book/) its a bit of a confusing resource and improving [wasm-bindgen guide](https://rustwasm.github.io/docs/wasm-bindgen/) would be more useful. As to tooling I can [throw my hat into the ring](https://github.com/pauldorehill/cargo-wasm) & I'd like to help out updating docs once there is a clear path forward.

**Comment by alexcrichton** [comment]
I agree that at this point it's probably best to update wasm-bindgen's documentation. I probably wouldn't remove wasm-pack entirely just yet, but it should probably get all consolidated onto one location (e.g. one page) and everything else in the documentation talks about `wasm-bindgen`'s CLI interface instead. If you'd be willing to help out with that then that would be great!

For repo movement it's never felt quite right to move just yet. I don't feel like there's a strong detriment being where it is right now, although it's not really a bonus either. I think that it probably makes more sense to figure out what to do with everything in this org all at once rather than one at a time, or at least that's what I hope.

**Comment by hamza1311** [comment]
Great, I'll go throw PRs and see what I can review and approve them. I'll let you know when a PR is ready to be merged
